experimental/uefi/runtime/src/lib.rs

```diff
@@ -52,24 +52,36 @@ impl<T: Channel> Framed<T> {
     }
 
     pub fn read_frame(&mut self) -> anyhow::Result<Frame> {
-        let mut length_buf = [0; FRAME_LENGTH_ENCODED_SIZE];
-        self.inner.recv(&mut length_buf)?;
-        let length = FrameLength::from_be_bytes(length_buf);
-        let mut body: Vec<u8> = vec![0; length.into()];
-        self.inner.recv(&mut body)?;
+        let length = {
+            let mut length_bytes = [0; FRAME_BYTES_SIZE];
+            self.inner.recv(&mut length_bytes)?;
+            FrameLength::from_le_bytes(length_bytes)
+        };
+
+        let body = {
+            let body_length: u32 = length - FRAME_BYTES_SIZE as u32;
+            let mut body: Vec<u8> = vec![
```

Comment:
Does it have to be initialised to 0, or is `Vec::with_capacity` sufficeiont here?

Reply:
Unfortunately it does need to be initialised, since our receive function fills the length of vector.